asv_bench/benchmarks/dataset_io.py

```diff
@@ -476,3 +508,145 @@ def cleanup(self):
 
     def time_write(self):
         self.write.compute()
+
+
+class IOReadSingleFile(IOSingleNetCDF):
+    def setup(self, *args, **kwargs):
+        self.make_ds()
+
+        self.filepaths = {}
+        for engine in _ENGINES:
+            self.filepaths[engine] = f"test_single_file_with_{engine}.nc"
+            self.ds.to_netcdf(self.filepaths[engine], engine=engine)
+
+    @parameterized(["engine", "chunks"], (_ENGINES, [None, {}]))
+    def time_read_dataset(self, engine, chunks):
+        xr.open_dataset(self.filepaths[engine], engine=engine, chunks=chunks)
+
+
+class IOReadCustomEngine:
```

Comment:
Thanks this is a great benchmark.

Shall we stick this in `xarray.tests` instead? I'm not sure if we have something similar for our tests already.